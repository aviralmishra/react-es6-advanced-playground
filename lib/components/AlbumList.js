import React from 'react';

import Album from './Album';

const AlbumList = (props) => {
  return (
    <div>
      {
        Object.values(props.albums).map(album => {
          return <Album key={album.id} album={album}/>;
        })
      }
    </div>
  );
};

export default AlbumList;
