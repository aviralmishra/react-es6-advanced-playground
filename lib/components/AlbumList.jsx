import React from 'react';

import Album from './Album.jsx';

const AlbumList = (props) => {
  return (<div>
    {
      Object.values(props.albums).map(album => {
        return <Album key={album.id} album={album} actions={props.albumActions}/>;
      })
    }
  </div>);
};

export default AlbumList;
