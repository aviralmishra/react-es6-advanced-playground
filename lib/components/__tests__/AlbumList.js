import React from 'react';

import AlbumList from '../AlbumList';

import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

describe('Album list', () => {
  const testProps = {
    'albums': {
      'album-1001': {
        'id': 'album-1001',
        'title': 'We Were Here',
        'description': 'We Were Here is the first album by Joshua Radin.',
        'released': '2006-06-13',
        'artistId': 'artist-1001'
      },
      'album-1002': {
        'id': 'album-1002',
        'title': 'Simple Times',
        'description': 'Simple Times is the second album by Joshua Radin.',
        'released': '2008-09-09',
        'artistId': 'artist-1002'
      }
    }
  };

  it('renders correctly', () => {
    const wrapper = shallow(<AlbumList {...testProps}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('AlbumContainer').length).toBe(2);
  });
});
