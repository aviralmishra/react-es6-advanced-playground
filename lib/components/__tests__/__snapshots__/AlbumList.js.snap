// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album list renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlbumList
    albums={
      Object {
        "album-1001": Object {
          "artistId": "artist-1001",
          "description": "We Were Here is the first album by Joshua Radin.",
          "id": "album-1001",
          "released": "2006-06-13",
          "title": "We Were Here",
        },
        "album-1002": Object {
          "artistId": "artist-1002",
          "description": "Simple Times is the second album by Joshua Radin.",
          "id": "album-1002",
          "released": "2008-09-09",
          "title": "Simple Times",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AlbumContainer
          album={
            Object {
              "artistId": "artist-1001",
              "description": "We Were Here is the first album by Joshua Radin.",
              "id": "album-1001",
              "released": "2006-06-13",
              "title": "We Were Here",
            }
          }
        />,
        <AlbumContainer
          album={
            Object {
              "artistId": "artist-1002",
              "description": "Simple Times is the second album by Joshua Radin.",
              "id": "album-1002",
              "released": "2008-09-09",
              "title": "Simple Times",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "album-1001",
        "nodeType": "class",
        "props": Object {
          "album": Object {
            "artistId": "artist-1001",
            "description": "We Were Here is the first album by Joshua Radin.",
            "id": "album-1001",
            "released": "2006-06-13",
            "title": "We Were Here",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "album-1002",
        "nodeType": "class",
        "props": Object {
          "album": Object {
            "artistId": "artist-1002",
            "description": "Simple Times is the second album by Joshua Radin.",
            "id": "album-1002",
            "released": "2008-09-09",
            "title": "Simple Times",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AlbumContainer
            album={
              Object {
                "artistId": "artist-1001",
                "description": "We Were Here is the first album by Joshua Radin.",
                "id": "album-1001",
                "released": "2006-06-13",
                "title": "We Were Here",
              }
            }
          />,
          <AlbumContainer
            album={
              Object {
                "artistId": "artist-1002",
                "description": "Simple Times is the second album by Joshua Radin.",
                "id": "album-1002",
                "released": "2008-09-09",
                "title": "Simple Times",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "album-1001",
          "nodeType": "class",
          "props": Object {
            "album": Object {
              "artistId": "artist-1001",
              "description": "We Were Here is the first album by Joshua Radin.",
              "id": "album-1001",
              "released": "2006-06-13",
              "title": "We Were Here",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "album-1002",
          "nodeType": "class",
          "props": Object {
            "album": Object {
              "artistId": "artist-1002",
              "description": "Simple Times is the second album by Joshua Radin.",
              "id": "album-1002",
              "released": "2008-09-09",
              "title": "Simple Times",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
